### PR TITLE
Fix request dep

### PR DESCRIPTION
**Pull request by firedog1024**
Switched out the request-promise-native NPM module that is fully deprecated for node-fetch NPM module.  Changed code in index.js and engine.js to use new library.  Changed out the package.json for the new library.  Rebuilt the deployment zip file.  I changed the deployment manifest to point at my branch so I could do a deployment test then reverted that change back to pointing at master.

### CODE REVIEW

File: IoTCIntegration/lib/engine.js [hunk 1]
```diff
@@ -111,26 +111,24 @@ async function getDeviceHub(context, device) {
...
-        url: `https://${registrationHost}/${context.idScope}/registrations/${deviceId}/register?api-version=${registrationApiVersion}`,
         method: 'PUT',
-        json: true,
-        headers: { Authorization: sasToken },
-        body: { registrationId: deviceId, payload: { iotcModelId: device.modelId } }
+        headers: { 'Content-Type': 'application/json', Authorization: sasToken },
+        body: JSON.stringify({ registrationId: deviceId, payload: { iotcModelId: device.modelId } })
     };
-
+    context.log(registrationOptions);
```
Comment (peolivei2) [inline IoTCIntegration/lib/engine.js:120]: Please remove this debug log

File: IoTCIntegration/lib/engine.js [hunk 2]
```diff
@@ -111,26 +111,24 @@ async function getDeviceHub(context, device) {
...
     try {
         context.log('[HTTP] Initiating device registration');
-        const response = await request(registrationOptions);
-
+        const response = await fetch(url, registrationOptions).then(res => res.json());
+        context.log(response);
```
Comment (peolivei2) [inline IoTCIntegration/lib/engine.js:124]: Remove this one as well

File: IoTCIntegration/package.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "azure-iot-device": "^1.7.4",
     "azure-iot-device-http": "^1.7.4",
     "request": "^2.88.0",
```
Comment (peolivei2) [inline IoTCIntegration/package.json:10]: Please remove request as well. It's the source of the deprecation.

**Comment by peolivei2** [comment]
Also update the package hash in the README

File: IoTCIntegration/lib/engine.js (hunk 1 above)
Comment (firedog1024) [reply IoTCIntegration/lib/engine.js]: Done

File: IoTCIntegration/lib/engine.js (hunk 2 above)
Comment (firedog1024) [reply IoTCIntegration/lib/engine.js]: Done

File: IoTCIntegration/package.json (hunk 3 above)
Comment (firedog1024) [reply IoTCIntegration/package.json]: Done

**Comment by firedog1024** [comment]
Updated the zip file with the changed files.  Got the SHA256 CRC value and amended the readme.md

**Pull request merged**